consensus/polybft/consensus_runtime.go

```diff
@@ -1328,6 +1315,38 @@ func (c *consensusRuntime) BuildRoundChangeMessage(
 	return signedMsg
 }
 
+func (c *consensusRuntime) updateProposerSnapshot(snapshot *ProposerSnapshot, blockNumber uint64) error {
+	const saveEveryNIterations = 5
+
+	c.logger.Info("Update proposal snapshot started", "block", blockNumber)
+
+	from := snapshot.Height
+
+	for height := from; height <= blockNumber; height++ {
+		c.config.proposerCalc.Update(snapshot, height, c.config)
+
+		// write snapshot every saveEveryNIterations iterations
+		// this way, we prevent data loss on long calculations
+		if (height-from+1)%saveEveryNIterations == 0 {
+			if err := c.state.writeProposerCalculatorSnapshot(snapshot); err != nil {
+				return fmt.Errorf("cannot save proposer calculator snapshot for block %d: %w", height, err)
+			}
+		}
+	}
+
+	// write snapshot if not already written
+	if (blockNumber-from+1)%saveEveryNIterations != 0 {
+		if err := c.state.writeProposerCalculatorSnapshot(snapshot); err != nil {
+			return fmt.Errorf("cannot save proposer calculator snapshot for block %d: %w", blockNumber, err)
+		}
+	}
+
+	// requires Lock
```

Comment:
Is this comment necessary?

Reply:
Will be changed with https://github.com/0xPolygon/polygon-edge/pull/1047